docs/core/extensions/file-globbing.md

```diff
@@ -0,0 +1,248 @@
+
+```csharp
+Matcher matcher = new();
+matcher.AddExcludePatterns(new [] { "*.txt", "*.asciidoc", "*.md" });
+```
+
+This extension method iterates over all of the provided patterns calling <xref:Microsoft.Extensions.FileSystemGlobbing.Matcher.AddExclude%2A> on your behalf.
+
+### Multiple inclusions
+
+To add multiple include patterns you can use:
+
+```csharp
+Matcher matcher = new();
+matcher.AddInclude("*.txt");
+matcher.AddInclude("*.asciidoc");
+matcher.AddInclude("*.md");
+```
+
+Alternatively, you can use the <xref:Microsoft.Extensions.FileSystemGlobbing.MatcherExtensions.AddIncludePatterns(Microsoft.Extensions.FileSystemGlobbing.Matcher,System.Collections.Generic.IEnumerable{System.String}[])?displayProperty=nameWithType> to add multiple include patterns in a single call:
+
+```csharp
+Matcher matcher = new();
+matcher.AddIncludePatterns(new[] { "*.txt", "*.asciidoc", "*.md" });
+```
+
+This extension method iterates over all of the provided patterns calling <xref:Microsoft.Extensions.FileSystemGlobbing.Matcher.AddInclude%2A> on your behalf.
+
+### Get all matching files
+
+To get all matching files, you have to call <xref:Microsoft.Extensions.FileSystemGlobbing.Matcher.Execute(Microsoft.Extensions.FileSystemGlobbing.Abstractions.DirectoryInfoBase)?displayProperty=nameWithType> either directly or indirectly. To call it directly you need a search directory:
+
+```csharp
+Matcher matcher = new();
+matcher.AddIncludePatterns(new[] { "*.txt", "*.asciidoc", "*.md" });
+
+string searchDirectory = "../starting-folder/";
+
+PatternMatchingResult result = matcher.Execute(
+    new DirectoryInfoWrapper(
+        new DirectoryInfo(searchDirectory)));
+
+// Use result.HasMatches and results.Files.
+// The files in the results object are file paths relative to the search directory.
+```
+
+The preceding C# code:
+
+- Instantiates a <xref:Microsoft.Extensions.FileSystemGlobbing.Matcher> object.
+- Calls <xref:Microsoft.Extensions.FileSystemGlobbing.MatcherExtensions.AddIncludePatterns(Microsoft.Extensions.FileSystemGlobbing.Matcher,System.Collections.Generic.IEnumerable{System.String}[])> to add several file name patterns to include.
+- Declares and assigns the search directory value.
+- Instantiates a <xref:System.IO.DirectoryInfo> from the given `searchDirectory`.
+- Instantiates a <xref:Microsoft.Extensions.FileSystemGlobbing.Abstractions.DirectoryInfoWrapper> from the `DirectoryInfo` it wraps.
+- Calls `Execute` given the `DirectoryInfoWrapper` instance to yield a <xref:Microsoft.Extensions.FileSystemGlobbing.PatternMatchingResult> object.
+
+> [!NOTE]
+> The `DirectoryInfoWrapper` type is defined in the `Microsoft.Extensions.FileSystemGlobbing.Abstractions` namespace, and the `DirectoryInfo` type is defined in the `System.IO` namespace. To avoid unnecessary `using` statements, you can use the provided extension methods.
```

Comment:
It's unclear to me what namespace I'm avoiding, and with which extension methods. I wonder if this alert box is really necessary -- would someone really choose an extension method for the purposes of avoiding a using?

Reply:
I added it because they avoided to `using` statements, which to me is rather compelling. They remove some inner details that are less relevant to the common consumer.